File: app/src/main/java/id/bdp/baseapp/ui/splash/SplashActivity.java
package id.bdp.baseapp.ui.splash;

import android.os.Bundle;

import id.bdp.baseapp.R;
import id.bdp.baseapp.base.BaseActivity;

public class SplashActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
    }

//    private void showDialogGeneralInfo(String type) {
//        View dialogView = getLayoutInflater().inflate(R.layout.dialog_info_general, null);
//        dialogInfoGeneralFacility = DialogHelper.getInstance().initCustomDialog(this, dialogView);
//
//        final TextView tvTitle = dialogInfoGeneralFacility.findViewById(R.id.tv_title);
//        final TextView tvContent = dialogInfoGeneralFacility.findViewById(R.id.tv_content);
//        MaterialRippleLayout mrClose = dialogInfoGeneralFacility.findViewById(R.id.mr_close);
//
//        tvTitle.setText((type.contains("general")) ? getString(R.string.title_info_general_facilities) : (type.contains("room")) ? getString(R.string.title_info_room_facilities) : getString(R.string.title_info_add_cost));
//        tvContent.setText((type.contains("general")) ? getString(R.string.detail_content_info_general_facilities) : (type.contains("room")) ? getString(R.string.detail_content_info_room_facilities) : getString(R.string.detail_content_info_add_cost));
//
//        mrClose.setOnClickListener(v -> dialogInfoGeneralFacility.dismiss());
//
//        dialogInfoGeneralFacility.show();
//    }
}

File: app/src/main/java/id/lima/moviecatalog/ui/splash/SplashActivity.kt
package id.lima.moviecatalog.ui.splash

import android.content.Context
import android.content.Intent
import android.os.Bundle
import id.lima.moviecatalog.R
import id.lima.moviecatalog.base.BaseActivity
import id.lima.moviecatalog.ui.main.MainActivity
import id.lima.moviecatalog.util.GlobalHelper

class SplashActivity : BaseActivity(), SplashContract.View {

    var presenter: SplashContract.Presenter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        presenter?.moveToNext(Intent(context, MainActivity::class.java))
    }

    override fun showNetworkNotAvailableError() {
        GlobalHelper.getInstance().makeToast(getString(R.string.connection_error), this)
    }

    override fun getContext(): Context {
        return this
    }
}